Ignore `mutable-argument-default (B006)` in `.pyi` stubs

[Mutable argument defaults](https://docs.astral.sh/ruff/rules/mutable-argument-default/) can only be a problem at runtime. But at the moment, `B006` will also be reported in `.pyi` stub files, even though it is certain that mutations can never occur there.

For example, consider the following function in [`scipy-stubs/stats/_mstats_basic.pyi:578`](https://github.com/jorenham/scipy-stubs/blob/5a2603e9fc3c37e9355773da31d294ac688d699d/scipy-stubs/stats/_mstats_basic.pyi#L578-L585):

```pyi
def mquantiles(
    a: _ArrayLikeFloat_co,
    prob: _ArrayLikeFloat_co = [0.25, 0.5, 0.75],
    alphap: AnyReal = 0.4,
    betap: AnyReal = 0.4,
    axis: CanIndex | None = None,
    limit: tuple[AnyReal, AnyReal] | tuple[()] = (),
) -> _MArrayND: ...
```

For the `prob` parameter the following error is reported:

```
scipy-stubs/stats/_mstats_basic.pyi:580:32: B006 Do not use mutable data structures for argument defaults
```

I think that "mutable" defaults should be allowed in cases like these. Having explicit parameter defaults that match the runtime can be seen as a form of documentation. This is especially useful when the runtime sources aren't available, which is quite common in this [`scipy-stubs`](https://github.com/jorenham/scipy-stubs) case.

--- a/crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B006_1.pyi
+++ b/crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B006_1.pyi
@@ -1,0 +1,10 @@
+# we disable this rule for pyi files
+
+def mquantiles(
+    a: _ArrayLikeFloat_co,
+    prob: _ArrayLikeFloat_co = [0.25, 0.5, 0.75],
+    alphap: AnyReal = 0.4,
+    betap: AnyReal = 0.4,
+    axis: CanIndex | None = None,
+    limit: tuple[AnyReal, AnyReal] | tuple[()] = (),
+) -> _MArrayND: ...

--- a/crates/ruff_linter/src/rules/flake8_bugbear/mod.rs
+++ b/crates/ruff_linter/src/rules/flake8_bugbear/mod.rs
@@ -43,6 +43,7 @@ mod tests {
     #[test_case(Rule::MutableArgumentDefault, Path::new("B006_7.py"))]
     #[test_case(Rule::MutableArgumentDefault, Path::new("B006_8.py"))]
     #[test_case(Rule::MutableArgumentDefault, Path::new("B006_B008.py"))]
+    #[test_case(Rule::MutableArgumentDefault, Path::new("B006_1.pyi"))]
     #[test_case(Rule::NoExplicitStacklevel, Path::new("B028.py"))]
     #[test_case(Rule::RaiseLiteral, Path::new("B016.py"))]
     #[test_case(Rule::RaiseWithoutFromInsideExcept, Path::new("B904.py"))]

--- a/crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs
+++ b/crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs
@@ -86,6 +86,11 @@ impl Violation for MutableArgumentDefault {
 
 /// B006
 pub(crate) fn mutable_argument_default(checker: &mut Checker, function_def: &ast::StmtFunctionDef) {
+    // Skip stub files
+    if checker.source_type.is_stub() {
+        return;
+    }
+
     for ParameterWithDefault {
         parameter,
         default,

